core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1375,54 +1397,85 @@ public void rollGeneration() throws IOException {
         }
     }
 
-    @Override
-    public long prepareCommit() throws IOException {
+    /**
+     * Prepares a translog commit by setting the current committing generation and rolling the translog generation.
+     *
+     * @throws IOException if an I/O exception occurred while rolling the translog generation
+     */
+    public void prepareCommit() throws IOException {
```

Comment:
now that we don't use the 2phase commit interface anymore we can rename these methods IMO